### PR TITLE
Update: Octane filter builders

### LINKED ISSUE #782: Reports components/filter-builders

### PR DESCRIPTION
**Pull request by kevinhinterlong**
Resolves #782 

## Description
Convert filter builder components to octane

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/filter-builders/base.js [hunk 1]
```diff
@@ -27,42 +23,47 @@ export default Component.extend({
    * @property {String} filter.operator.valuesComponent - name of component to use for selecting filter values
    * @property {Array} filter.values
    */
-  filter: undefined,
+  get filter() {
```
Comment (jkusa) [inline packages/reports/addon/components/filter-builders/base.js:26]: does this need to be a getter?

File: packages/reports/addon/components/filter-builders/base.js [hunk 2]
```diff
@@ -27,42 +23,47 @@ export default Component.extend({
...
    * @property {String} displayName - display name for the filter
    */
-  displayName: readOnly('filter.subject.name'),
+  @readOnly('filter.subject.name') displayName;
 
   /**
    * @property {Array} supportedOperators - list of valid values for filter.operator
    */
-  supportedOperators: computed(function() {
+  @computed
```
Comment (jkusa) [inline packages/reports/addon/components/filter-builders/base.js:38]: `supportedOperators = []`?

File: packages/reports/addon/components/filter-builders/date-dimension.js [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
-  requestFragment: undefined,
+  requestFragment = undefined;
 
   /**
    * @property {Array} supportedOperators
    * @override
    */
-  supportedOperators: computed(function() {
+  @computed
+  get supportedOperators() {
```
Comment (jkusa) [inline packages/reports/addon/components/filter-builders/date-dimension.js:26]: literal?

File: packages/reports/addon/components/filter-builders/metric.js [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
-
+@templateLayout(layout)
+@tagName('')
+class MetricFilterBuilderComponent extends BaseFilterBuilderComponent {
   /**
    * @property {Object[]} supportedOperators
    */
-  supportedOperators: computed(function() {
+  @computed
+  get supportedOperators() {
```
Comment (jkusa) [inline packages/reports/addon/components/filter-builders/metric.js:25]: same

File: packages/reports/addon/components/filter-builders/metric.js [hunk 5]
```diff
@@ -63,26 +65,28 @@ export default Base.extend({
...
+  requestFragment = undefined;
 
   /**
    * @property {String} displayName - display name for the filter with metric and parameters
    */
-  displayName: computed('filter.subject.{metric,parameters}', function() {
+  @computed('filter.subject.{metric,parameters}')
+  get displayName() {
     let metric = get(this, 'filter.subject');
     return metricFormat(metric, metric.metric?.name);
```
Comment (jkusa) [inline packages/reports/addon/components/filter-builders/metric.js:81]:
```suggestion
    const metric = this.filter.subject;
    return metricFormat(metric, metric.metric?.name);
```

File: packages/reports/addon/components/filter-builders/base.js (hunk 1 above)
Comment (kevinhinterlong) [reply packages/reports/addon/components/filter-builders/base.js:26]: I can try it with just `filter;`, but `filter = undefined` results in an error. I think it's trying to do `this.filter = undefined` during the constructor but in subclasses we have `get filter()` already defined, so it thinks we are trying to set a getter only property

Comment (kevinhinterlong) [reply packages/reports/addon/components/filter-builders/base.js:26]: Ok coming back to this, if we do
```
filter = undefined; // breaks
filter; // breaks
// filter; // works, but confusing
get filter() { // works, slightly less confusing
  return undefined;
}
```

basically the same thing with `supportedOperators` so I'm opting to leave it as is

**Pull request merged**